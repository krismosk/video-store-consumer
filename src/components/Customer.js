import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './RentalCard.css';

const Customer = ({ id, name, address, city, state, postalCode, phone, findCustomer }) => {
  
  return (
  <div>
    <section>
      <p>{ id }</p>
      <p>{ name }</p>
      <p>{ address }</p>
      <p>{ city }</p>
      <p>{ state }</p>
    </section>
    <button onClick={() => {findCustomer(id)}} type="button">Select</button>
  </div>
  )
}

Customer.propTypes = {
  findCustomer: PropTypes.func.isRequired,
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  address: PropTypes.string,
  city: PropTypes.string,
  state: PropTypes.string,
  postalCode: PropTypes.string,
  phone: PropTypes.string,
}

export default Customer;