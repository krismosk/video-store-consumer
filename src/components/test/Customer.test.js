import React from 'react';
import { render, cleanup, getByText } from '@testing-library/react';
import Customer from '../Customer';

const customer = <Customer 
  key={ 1 }
  id={ 1337 }
  name={ 'test customer' }
  address={ 'test address' }
  city={ 'test city' }
  state={ 'test state' }
  postalCode={ 'test postalcode' }
  phone={ '1111111111' }
  findCustomer={ () => { } }
/>;

describe ('Customer', () => {
  test('it can be rendered and matches the snapshot', () => {
    const { asFragment } = render( customer )

    expect(asFragment()).toMatchSnapshot();

    cleanup();
  });
})

