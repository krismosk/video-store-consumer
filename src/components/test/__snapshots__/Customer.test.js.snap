// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customer it can be rendered and matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <section>
      <p>
        1337
      </p>
      <p>
        test customer
      </p>
      <p>
        test address
      </p>
      <p>
        test city
      </p>
      <p>
        test state
      </p>
    </section>
    <button
      type="button"
    >
      Select
    </button>
  </div>
</DocumentFragment>
`;
